### PR TITLE
Add Entity Framework request culture provider

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

File: src/Modules/SimplCommerce.Module.Localization/Controllers/LocalizationController.cs [hunk 1]
```diff
@@ -10,11 +13,12 @@ public class LocalizationController : Controller
         [HttpPost]
         public IActionResult SetLanguage(string culture, string returnUrl)
         {
-            Response.Cookies.Append(
-                CookieRequestCultureProvider.DefaultCookieName,
-                CookieRequestCultureProvider.MakeCookieValue(new RequestCulture(culture)),
-                new CookieOptions { Expires = DateTime.UtcNow.AddYears(1) }
-            );
+            var cultureRepository = HttpContext.RequestServices.GetRequiredService<IRepositoryWithTypedId<Culture, string>>();
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.Localization/Controllers/LocalizationController.cs]: cultureRepository should be injected in the constructor

**Comment by thiennn** [comment]
With previous implementation, every users can set their own prefer culture, and it don't impact to other users. With this implementation, it seems that all the users will share the same only 1 culture.

File: src/Modules/SimplCommerce.Module.Localization/Controllers/LocalizationController.cs (hunk 1 above)
Comment (hishamco) [reply src/Modules/SimplCommerce.Module.Localization/Controllers/LocalizationController.cs]: Yep

**Comment by hishamco** [comment]
> With this implementation, it seems that all the users will share the same only 1 culture.

You are right, seems I need to change the logic little bit, so we can keep track of the culture for the authenticated user by adding a new `Culture` column in the `Users` table

`IsDefault` column is no longer needed in this case

**Comment by thiennn** [comment]
What is the drawback of the previous implementation?

**Comment by hishamco** [comment]
I'm refactoring the code, so the culture will be stored into the user preferences which did the same job, furthermore the cookie has expiration date, but IMHO the new provider will works smoothly aligned with `EfStringLocalizer` so both localization resources & cultures will be in the database
Let me update the code now and have a look to it

**Comment by thiennn** [comment]
There can be a potential confusion for users after they signed in. I mean the culture change after they signed in. Because the guest user and the signed user are different. I think we should hook in to the UserSignedIn event to do some logic as https://github.com/simplcommerce/SimplCommerce/blob/master/src/Modules/SimplCommerce.Module.ShoppingCart/Events/UserSignedInHandler.cs

**Comment by hishamco** [comment]
FYI I didn't use the cookies at all, I knew I was mentioned in previous thread, after that I realize it's confusing, so I used the `IWorkContext.GetCurrentUser()` so each user will have a culture based on his/her preferences
Hope you have a look to the updated PR

**Comment by thiennn** [comment]
I completely understand your code. Maybe my previous comment is hard to understand. Please read it again and ping me if you still not get my point

**Comment by hishamco** [comment]
Got it!! I will use the `UserSignedInHandler` now

**Comment by hishamco** [comment]
@thangln1003 final review

**Pull request merged**